### PR TITLE
Add a CONTRIBUTING.md

### LINKED ISSUE #40: Create CONTRIBUTING.md
**Background**

As cladetime has evolved from a package primarily used for backend pipelines to one that includes a user-facing API, it's time to revisit its local dev setup instructions.

**Definition of done**

- [x] Move project dev setup instructions from the main README to a new CONTRIBUTING.md file.
- [x] Move to a uv-driven workflow (for ease of use, and so we can try it out)
- [x] Include instructions for locally testing our readthedocs components

### PR DESCRIPTION
**Pull request by bsweger**
Closes #40 

Add a CONTRIBUTING.md file to Cladetime

This is notably lacking code of conduct information, but happy to
add that in if the Reich Lab has one. Other than that, this
is hopefully a good foundation to iterate on.

### CODE REVIEW

File: requirements/requirements-doc.txt
Comment (bsweger) [inline requirements/requirements-doc.txt:1]: A new requirements file that is specific to documentation. Most people won't need this unless they want to build and serve the docs locally.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+### Adding new dependencies
+
+If your change requires a new dependency in the project, add it as follows:
+
+1. Add the dependency to the `dependencies` section of [`pyproject.toml`](pyproject.toml).
+2. Update the requirements files:
+
+    ```bash
+    uv pip compile pyproject.toml -o requirements/requirements.txt && uv pip compile pyproject.toml --extra dev -o requirements/requirements-dev.txt
+    ```
```
Comment (elray1) [inline CONTRIBUTING.md:118]: should this include stuff for how to build `requirements/requirements-doc.txt`?

Comment (bsweger) [reply CONTRIBUTING.md:118]: Good question--I was operating under the assumption that the docs requirements will be pretty stable and contributors won't have to worry about them (I know, I know)

Comment (bsweger) [reply CONTRIBUTING.md:118]: @elray1 I added that info to the documentation section

**Review by elray1 — APPROVED** [review]
lgtm!

**Pull request merged**